doc/controller-card_poweroff.md

```diff
@@ -0,0 +1,143 @@
+/components/component[name=controller0]/controller-card/state/power-admin-state, POWER_ENABLED
+/components/component[name=controller1]/state/redundant-role, SECONDARY
+/components/component[name=controller1]/controller-card/config/power-admin-state, POWER_DISABLED
+/components/component[name=controller1]/controller-card/state/power-admin-state, POWER_DISABLED
+/components/component[name=controller1]/state/last-poweroff-reason/trigger, USER_INITIATED
+/components/component[name=controller1]/state/last-poweroff-reason/details, "User initiated Shutdown"
+/components/component[name=controller1]/state/last-poweroff-time, 1706590714681765937
+```
+
+### Scenario 2 - Power off primary CONTROLLER_CARD
+    
+If controller-card0 is PRIMARY and controller-card1 is SECONDARY and if controller-card0 is set to config/power-admin-state = POWER_DISABLED by an operator, then controller-card0 will stay powered-on until the next reboot. `state/power-admin-state` must show as POWER_ENABLED. If a reboot of the PRIMARY CONTROLLER_CARD occurs,  `state/last-poweroff-time` should record the time when the card powers-off, `state/last-poweroff-reason/trigger` should show as USER_INITIATED and `/state/last-poweroff-reason/details` may be updated. For example: 
+
+When controller-card0 is PRIMARY:
+
+```          
+/components/component[name=controller0]/state/redundant-role, PRIMARY
+/components/component[name=controller0]/controller-card/config/power-admin-state, POWER_DISABLED
+/components/component[name=controller0]/controller-card/state/power-admin-state, POWER_ENABLED
+/components/component[name=controller1]/state/redundant-role, SECONDARY
+/components/component[name=controller1]/controller-card/state/power-admin-state, POWER_ENABLED
+```
+
+After controller-card0 transitions to redundant-role SECONDARY:
+
+```
+/components/component[name=controller0]/state/redundant-role, SECONDARY
+/components/component[name=controller0]/controller-card/config/power-admin-state, POWER_DISABLED
+/components/component[name=controller0]/controller-card/state/power-admin-state, POWER_DISABLED
+/components/component[name=controller0]/state/last-poweroff-reason/trigger, USER_INITIATED
+/components/component[name=controller0]/state/last-poweroff-reason/details, "User initiated Shutdown"
+/components/component[name=controller0]/state/last-poweroff-time, 1706590714681765937
+/components/component[name=controller1]/state/redundant-role, PRIMARY
+/components/component[name=controller1]/controller-card/config/power-admin-state, POWER_ENABLED
+/components/component[name=controller1]/controller-card/state/power-admin-state, POWER_ENABLED
+```
+
+2. A controller-card which is in state/redundant-role=SECONDARY and is config/power-admin-state=POWER_DISABLED should remain powered off, even after a reboot. It is possible that after a reboot, both the controller cards are powered-on. However as soon as the configuration is loaded, the system should power-off the subject controller-card.
```

Comment:
Perhaps the term "reboot" is used somewhat ambiguously throughout the text, but here it is stated that the card _should_ remain powered off, but may be powered-on temporarily. And at the end of the document, in the scenario # 8, you provide an example where a reboot (through power cycling) must initiate a new boot cycle.

>If the device is rebooted (for example via externally removing and reapplying power ), what should happen?
>    If during reboot the device continues to have a single controller-card, then Rule#5 above should be followed.

Reply:
To increase clarity, I recommend replacing 
`It is possible that after a reboot, both the controller cards are powered-on` with 
`During a cold-boot, some implementations may power up both CONTROLLER_CARDS.`

Reply:
Why only cold boot? We are not asking implementations to maintain state even during warm boot. Therefore it is possible that both controller-cards may turn-on simultaneously post a reboot (cold or warm). Expectations are that the implementation loads the configuration on both cards, the card w/ `power-admin-state=POWER_DISABLED` will be powered-off unless it is the only card.

Reply:
In my view, the ambiguity is that in order to follow the rules that you specified, e.g. the one I quoted:

>If the device is rebooted (for example via externally removing and reapplying power ), what should happen?
If during reboot the device continues to have a single controller-card, then Rule#5 above should be followed.

there must be a way to "reboot" (in your terms) the card that was previously powered-off via the config, and it cannot stay powered off during that time.

But rule 2 says otherwise:
> A controller-card which is in state/redundant-role=SECONDARY and is config/power-admin-state=POWER_DISABLED should remain powered off, even after a reboot.

if the card remains powered off, it will never know if its a single card in the system, or there's another card. So it cannot remain powered off.

Reply:
As per the offline discussions, @LimeHat seems like the requirement is understood. How an implementation implements the change is left on them.